uring/src/main/scala/fs2/io/uring/net/UringDatagramSocket.scala

```diff
@@ -117,9 +114,24 @@ private[net] object UringDatagramSocket {
   def apply[F[_]](ring: Uring[F], fd: Int)(implicit
       F: Async[F]
   ): Resource[F, UringDatagramSocket[F]] =
-    ResizableBuffer(65535).evalMap { buf =>
-      (Mutex[F], Mutex[F]).mapN(new UringDatagramSocket(ring, fd, buf, 65535, _, _))
-    }
+    ResizableBuffer(65535)
+      .evalMap { buf =>
+        (Mutex[F], Mutex[F]).mapN { (readMutex, writeMutex) =>
+          val iov = malloc(sizeof[uio.iovec]).asInstanceOf[Ptr[uio.iovec]]
+          val remoteAddr = malloc(sizeof[sockaddr_storage]).asInstanceOf[Ptr[sockaddr_storage]]
+          val msg = malloc(sizeof[msghdr]).asInstanceOf[Ptr[msghdr]]
```

Comment:
`malloc` may return `null` I think, we should check for that.

Reply:
Here's an example:
https://github.com/armanbilge/fs2-io_uring/blob/5af6a81284fafefd5747f26157f54c1b68b4b536/uring/src/main/scala/fs2/io/uring/net/ResizableBuffer.scala#L52-L54

Reply:
Would it be a good idea to make a more generic function in utils that allocates memory size given and check for null ?

Reply:
Yes, definitely! We it can return a `Resource[F, Ptr[Byte]]` that frees itself on release.